### DISCUSSION TITLE
[2022/06/28] Frontend Night

### DISCUSSION THREAD

**Discussion by jjenko** [wantedly/frontend-night#211] [Archive]
## Policy

- 資料の準備をがんばらない
- 参加・途中退席は自由
- スキップしない

話したい・聞きたいネタを書いてく ✏️

ハッシュタグ [#frontend_night](https://twitter.com/hashtag/frontend_night?src=hash)

Prev: https://github.com/wantedly/frontend-night/discussions/210 

 _Created from https://github.com/wantedly/frontend-night/discussions/179 by [issue-creator](https://github.com/rerost/issue-creator)_

**Comment by chloe463** [comment]
[JSer.Info](https://jser.info/2022/06/28/ecmascript-2022-typescript-4.8-beta-codemirror-6/)

**Comment by prsdta** [comment]
https://github.com/petyosi/react-virtuoso ?

**Reply by kobayang** [reply]
virtualization のライブラリ。可変サイズのコンポーネントが簡単に virtualization できる？

**Comment by kobayang** [comment]
chromatic お触り会してみた
https://www.chromatic.com/

**Reply by kobayang** [reply]
- コメントがつけられるのでcollaborationができそう
- 認証つきのstorybook もついてくる

**Comment by prsdta** [comment]
https://beta.reactjs.org/learn/you-might-not-need-an-effect

**Reply by kobayang** [reply]
> Recap
> - If you can calculate something during render, you don’t need an Effect.
> - To cache expensive calculations, add useMemo instead of useEffect.
> - To reset the state of an entire component tree, pass a different key to it.
> - To reset a particular bit of state in response to a prop change, set it during rendering.
> - Code that needs to run because a component was displayed should be in Effects, the rest should be in events.
> - If you need to update the state of several components, it’s better to do it during a single event.
> - Whenever you try to synchronize state variables in different components, consider lifting state up.
> - You can fetch data with Effects, but you need to implement cleanup to avoid race conditions.

**Comment by kobayang** [comment]
https://overreacted.io/ja/algebraic-effects-for-the-rest-of-us/
